python/cudf/cudf/utils/ioutils.py

```diff
@@ -2099,3 +2095,96 @@ def _read_byte_ranges(
 
     for worker in workers:
         worker.join()
+
+
+def _get_remote_bytes_all(
+    remote_paths, fs, *, blocksize=_BYTES_PER_THREAD_DEFAULT
+):
+    if (
+        len(remote_paths) >= 8  # Heuristic to avoid fs.sizes
+        or max(sizes := fs.sizes(remote_paths)) <= blocksize
+    ):
+        # Don't bother braking up individual files
+        return fs.cat_ranges(remote_paths, None, None)
+    else:
+        # Construct list of paths, starts, and ends
+        paths, starts, ends = [], [], []
+        for i, remote_path in enumerate(remote_paths):
+            for j in range(0, sizes[i], blocksize):
+                paths.append(remote_path)
+                starts.append(j)
+                ends.append(min(j + blocksize, sizes[i]))
```

Comment:
```suggestion
        for remote_path, size in zip(remote_paths, sizes):
            for j in range(0, size, blocksize):
                paths.append(remote_path)
                starts.append(j)
                ends.append(min(j + blocksize, size))
```

Also (because it's fun :)) this should work too and comprehensions are nice, but is definitely less readable
```
paths, starts, ends = zip(*[(r, j, min(j + blocksize, s)) for r, s in zip(files, sizes) for j in range(0, s, blocksize)])
```

Reply:
Sure, I never start with nested list comprehension, but happy to use it once the "spelled-out" logic is clear to a reviewer :)